proxy/proxy.go

```diff
@@ -374,18 +394,29 @@ func handleCacheHit(cacheKey string, content []byte, logger *slog.Logger) func(w
 			}
 		} else if strings.Contains(cacheKey, "LicenseQuery") {
 			typename := "RouterEntitlementsResult"
-			if len(content) == 0 {
+			cacheResponse := JWTCacheEntry{}
+			err := json.Unmarshal(content, &cacheResponse)
+			if err != nil {
+				logger.Error("Failed to unmarshal JWT cache entry", "err", err)
+				http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+				return nil
+			}
+
+			jwtEntitlement := &Jwt{Jwt: string(cacheResponse.Jwt)}
+			if len(cacheResponse.Jwt) == 0 {
 				typename = "Unchanged"
+				jwtEntitlement = nil
 			}
+
 			response = &UplinkLicenseResponse{
 				Data: struct {
 					RouterEntitlements UplinkRouterEntitlements `json:"routerEntitlements"`
 				}{
 					RouterEntitlements: UplinkRouterEntitlements{
-						ID:              time.Now().UTC().String(),
+						ID:              cacheResponse.Expiration,
 						Typename:        typename,
-						MinDelaySeconds: 60,
-						Entitlement:     Jwt{Jwt: string(content)},
+						MinDelaySeconds: 10,
```

Comment:
This works for now but eventually we are going to want to move this to a config or start up param